core/lib/mno_enterprise/concerns/models/ability.rb

```diff
@@ -129,8 +129,11 @@ def impac_abilities(user)
     end
 
     can :manage_kpi, MnoEnterprise::Impac::Kpi do |kpi|
-      dashboard = kpi.dashboard
-      authorize! :manage_dashboard, dashboard
+      if kpi.widget.present?
+        authorize! :manage_widget, MnoEnterprise::Impac::Widget.find(kpi.widget.id)
```

Comment:
@ouranos I could not because :manage_widget needs widget.dashboard, and it is not fetched in the kpi.widget hash. Is there a way to include deeper nested attributes?